builds/e2e/upgrade-scenarios.yaml

```diff
@@ -0,0 +1,229 @@
+        # are left using older runtime images.
+        1.2.0_to_1.3_only_edge_daemon:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.0"
+          runtimeModulesVersion: "1.2.0"
+        1.2.9_to_1.3_only_edge_daemon:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.9"
+          runtimeModulesVersion: "1.2.9"
+        # TODO: Uncomment if/when the .NET test framework is updated to allow overriding
+        # the default 1.2 schema version used in the EH desired properties section
+        # of the deployment file. Until then, these scenarios will fail
+        # 1.1.0_to_1.3_only_edge_daemon:
+        #   edgeDaemonVersion: "release/1.3"
+        #   bootstrapAgentVersion: "1.1.0"
+        #   runtimeModulesVersion: "1.1.0"
+        # 1.1.8_to_1.3_only_edge_daemon:
+        #   edgeDaemonVersion: "release/1.3"
+        #   bootstrapAgentVersion: "1.1.8"
+        #   runtimeModulesVersion: "1.1.8"
+        # 1.1.13_to_1.3_only_edge_daemon:
+        #   edgeDaemonVersion: "release/1.3"
+        #   bootstrapAgentVersion: "1.1.13"
+        #   runtimeModulesVersion: "1.1.13"
+
+    timeoutInMinutes: 90
+
+    steps:
+    - template: templates/e2e-setup.yaml
+
+    # If edgeDaemonVersion != "release/1.3", remove the aziot-edge and
+    # aziot-identity-service packages from $(System.ArtifactsDirectory) and download the desired
+    # versions.
+    - bash: |
+        if [ $(edgeDaemonVersion) != "release/1.3" ]; then
+          pushd $(System.ArtifactsDirectory)/$(artifactName)
+          rm aziot-edge*.deb
+          rm aziot-identity-service*.deb
+          apt-get download aziot-edge='$(edgeDaemonVersion)-1'
+          is_version=`sudo apt depends aziot-edge='$(edgeDaemonVersion)-1' | sed -En "s/Depends: aziot-identity-service \(= ([0-9]+.[0-9]+.[0-9]+-[0-9]+)\)/\1/p"`
+          apt-get download aziot-identity-service=`echo $is_version`
+          popd
+        fi
+      displayName: Replace Downloaded Packages
+
+    # If $bootstrapAgentVersion is set to 'release/1.3', the value of edgeAgentBootstrapImage in context.json 
+    # must be modified here so that it is not set to the default E2E test version (i.e. azureiotedge-agent-bootstrap-e2e-linux-amd64)
+    - bash: |
+        artifactInfoStr=`cat '$(System.ArtifactsDirectory)/$(az.pipeline.images.artifacts)/artifactInfo.txt'`
+        IFS='='
+        read -ra artifactInfo <<< "$artifactInfoStr"
+        imageId="${artifactInfo[1]}"
+        echo "imageId: ${imageId}"
+        cd $(Build.SourcesDirectory)/test/Microsoft.Azure.Devices.Edge.Test/bin/Debug/net*
+        echo "Original context.json:"
+        less context.json
+        if [ $(bootstrapAgentVersion) = "release/1.3" ]; then
+          sed -i 's/"edgeAgentBootstrapImage.*/"edgeAgentBootstrapImage": "edgebuilds.azurecr.io\/microsoft\/azureiotedge-agent:'${imageId}'",/' context.json
```

Comment:
I think it would be quite nice if this can be modify as pipeline variable so a dev can just test an upgrade scenario from god-knows-where-cr. (optional)

Reply:
Agreed, but let's save that for a future iteration.